rasterio/merge.py

```diff
@@ -22,6 +22,10 @@
 logger = logging.getLogger(__name__)
 
 
+class MergeError(RasterioError):
+    """Raised when rasters cannot be merged."""
```

Comment:
@snowman2 @vincentsarago @groutr this would be a little unusual for Rasterio, but what would you think about the exception being defined in the only module where it is used?

It prevents it from misuse. `except RasterioError` still catches it. The `merge()` tool isn't exported from `rasterio` anyway, so `from rasterio.merge import MergeError, merge` is simple enough.

Reply:
My initial reaction is -1, but there is likely something that I am missing. What misuse are you attempting to avoid?

Reply:
Exceptions in rasterio.errors may appear to be free for any purpose. An exception in merge.py signals that it's for that module specifically. You'd have to import the module to use it for an unrelated purpose. It's a small thing. I would probably design like this if I were starting a project from scratch.